igor-web/src/main/groovy/com/netflix/spinnaker/igor/config/TravisConfig.groovy

```diff
@@ -0,0 +1,110 @@
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import retrofit.Endpoints
+import retrofit.RequestInterceptor
+import retrofit.RestAdapter
+import retrofit.client.OkClient
+
+import javax.validation.Valid
+import java.util.concurrent.TimeUnit
+
+
+/**
+ * Converts the list of Travis Configuration properties a collection of clients to access the Travis hosts
+ */
+@Configuration
+@Slf4j
+@CompileStatic
+@ConditionalOnProperty("travis.enabled")
+class TravisConfig {
+
+    @Value('${client.timeout:30000}')
+    int clientTimeout
+
+    @Autowired(required = false)
+    BuildMasters buildMasters
+
+    @Bean
+    Map<String, TravisService> travisMasters(@Valid TravisProperties travisProperties) {
+        log.info "creating travisMasters"
+        Map<String, TravisService> travisMasters = (travisProperties?.masters?.collectEntries { TravisProperties.TravisHost host ->
+            log.info "bootstrapping ${host.address} as ${host.name}"
+            [(host.name): travisService(host.name, host.baseUrl, host.githubToken, travisClient(host.address, clientTimeout))]
+        })
+        buildMasters.map.putAll travisMasters
+        travisMasters
+    }
+
+    static TravisService travisService(String travisHostId, String baseUrl, String githubToken, TravisClient travisClient) {
+        return new TravisService(travisHostId, baseUrl, githubToken, travisClient)
+    }
+
+    static TravisClient travisClient(String address, int timeout = 30000) {
+        OkHttpClient client = new OkHttpClient()
+        client.setReadTimeout(timeout, TimeUnit.MILLISECONDS)
+
+        //Need this code because without FULL log level, fetching logs will fail. Ref https://github.com/square/retrofit/issues/953.
+        RestAdapter.Log fooLog = new RestAdapter.Log() {
+            @Override public void log(String message) {
+            }
+        }
+        RequestInterceptor requestInterceptor = address =~ "api.travis-ci.org" ? new TravisHeader() : new TravisEnterpriseHeader()
+        new RestAdapter.Builder()
+            .setEndpoint(Endpoints.newFixedEndpoint(address))
+            .setRequestInterceptor(requestInterceptor)
+            .setClient(new OkClient(client))
+            .setLog(fooLog)
+            .setLogLevel(RestAdapter.LogLevel.FULL)
+            .build()
+            .create(TravisClient)
+    }
+
+    static class TravisHeader implements RequestInterceptor {
+
+        @Override
+        void intercept(RequestInterceptor.RequestFacade request) {
+            request.addHeader("Accept", "application/vnd.travis-ci.2+json")
+            request.addHeader("User-Agent", "Travis/1.0")
```

Comment:
Why not Travis-Igor in that case? Just to be friendly to Travis, so they know where the traffic is coming from.